pkg/processor/build/builder_test.go

```diff
@@ -581,6 +668,86 @@ func (suite *testSuite) mergeDirectivesAndVerify(first map[string][]functionconf
 
 }
 
+func (suite *testSuite) testResolveFunctionPathRemoteCodeFile(fileExtension string) {
+
+	// mock http response
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	codeFileContent:= "some code..."
+	responder := func(req *http.Request) (*http.Response, error) {
+		responder := httpmock.NewStringResponder(200, codeFileContent)
+		response, err := responder(req)
+		response.ContentLength = int64(len(codeFileContent))
+		return response, err
+	}
+	pythonFileURL := "http://some-address.com/my-func."+fileExtension
+	httpmock.RegisterResponder("GET", pythonFileURL, responder)
+
+	// the python file will be downloaded here
+	err := suite.builder.createTempDir()
+	suite.NoError(err)
+
+	defer suite.builder.cleanupTempDir() // nolint: errcheck
+
+	path, err := suite.builder.resolveFunctionPath(pythonFileURL)
+	suite.NoError(err)
+
+	expectedFilePath := suite.builder.tempDir+"/download/my-func."+fileExtension
+	suite.Equal(expectedFilePath, path)
+
+	resultSourceCode, err := ioutil.ReadFile(expectedFilePath)
+	suite.Assert().NoError(err)
+
+	suite.Assert().Equal(codeFileContent, string(resultSourceCode))
+}
+
+func (suite *testSuite) testResolveFunctionPathArchive(buildConfiguration functionconfig.Build, archiveFileURL string) {
+	var destinationWorkDir string
+
+	suite.builder.options.FunctionConfig.Spec.Build = buildConfiguration
+
+	// the archive will be "downloaded" here
+	err := suite.builder.createTempDir()
+	suite.NoError(err)
+
+	defer suite.builder.cleanupTempDir() // nolint: errcheck
+
+	// mock the http/s3 response to be the test function archive file
+	if buildConfiguration.CodeEntryType != S3EntryType {
+		httpmock.Activate()
+		defer httpmock.DeactivateAndReset()
+		functionArchiveFileBytes, err := ioutil.ReadFile(FunctionsArchiveFilePath)
+
+		responder := func(req *http.Request) (*http.Response, error) {
+			response := &http.Response{
+				Status:        strconv.Itoa(200),
```

Comment:
use "200" or use `expectedStatusCode = 200` and use that variable wherever, no need to really strconv a magic number :)